Source/WebGPU/WGSL/Overload.h

```diff
@@ -74,13 +74,18 @@ using AbstractType = std::variant<
     Type*
 >;
 
+using AbstractScalarType = std::variant<
+    TypeVariable,
+    Type*
+>;
+
 struct AbstractVector {
-    TypeVariable element;
+    AbstractScalarType element;
```

Comment:
Why not use `AbstractType`? I know it's more permissive, but in practice I don't think it matters, and might even be more correct.

Reply:
I didn't use it because it was more permissive.